addons/l10n_es_edi_facturae/data/facturae_templates.xml

```diff
@@ -0,0 +1,439 @@
+<?xml version="1.0" encoding="utf-8"?>
+<odoo>
+    <data>
+        <!-- Sub-template used for every instance of AddressType -->
+        <template id="address_type">
+            <t t-if="address['CountryCode'] == 'ESP'">
+                <AddressInSpain>
+                    <Address t-out="address['Address']"/>
+                    <PostCode t-out="str(address['PostCode'])"/>
+                    <Town t-out="address['Town']"/>
+                    <Province t-out="address['Province']"/>
+                    <CountryCode t-out="address['CountryCode']"/>
+                </AddressInSpain>
+            </t>
+            <t t-else="">
+                <OverseasAddress>
+                    <Address t-out="address['Address']"/>
+                    <PostCodeAndTown t-out="' '.join(var for var in (address['PostCode'], address['Town']) if var)"/>
+                    <Province t-out="address['Province']"/>
+                    <CountryCode t-out="address['CountryCode']"/>
+                </OverseasAddress>
+            </t>
+        </template>
+
+        <!-- Sub-template used for every instance of ContactDetails -->
+        <template id="contact_details_type">
+            <Telephone t-if="ContactDetails.get('Telephone', False)" t-out="ContactDetails['Telephone']"/>
```

Comment:
All of these checks are useless I think.
```suggestion
            <Telephone t-out="ContactDetails.get('Telephone')"/>
```
will produce exactly the same output